### DISCUSSION TITLE
How to restrict which languages can be translated?

### DISCUSSION THREAD

**Discussion by probonopd** [WeblateOrg/weblate#6649] [Q&A]
Hello, is there a way in Hosted Weblate to restrict which languages can be translated? Our project is interested in only a couple of defined languages, but we are getting "random" ones, including ones we are not interested in.

I am sure this must be easy, but I didn't find it in the settings so far.

**Comment by probonopd** [comment]
Of course one can disable the automatic addition of translation files in each translation component, but that is rather cumbersome.

**Comment by nijel** [comment]
* You can disable adding new translations using https://docs.weblate.org/en/latest/admin/projects.html#adding-new-translation
* You can define regex filter for language code using https://docs.weblate.org/en/latest/admin/projects.html#language-filter

**Reply by probonopd** [reply]
Thanks @nijel - isn't this a bit cumbersome if you have many components in a project? What I am basically looking for is to define project-wide (for all components) which languages are desired.

**Reply by nijel** [reply]
This is currently per-component setting. You can automate that using API.

**Reply by probonopd** [reply]
Thanks for the clarification @nijel.
Maybe it can be considered as a future feature, I'd be surprised if I'm the only one who would like this.

**Reply by nijel** [reply]
In most cases, people address this by disabling adding new translations.

**Comment by nijel** [comment]
A few years later I've implemented this: https://github.com/WeblateOrg/weblate/pull/20959

**Comment by emmapeel2** [comment]
In my experience, when projects suffer of 'randomly added languages' they have usually marked the components' option `use as glossary`. This will generate a translation for each language used on the project.

**Comment by nijel** [comment]
https://github.com/WeblateOrg/weblate/pull/21501 turns this discussion into documentation update.
